src/Lecture2.hs

```diff
@@ -155,12 +176,87 @@ You're free to define any helper functions.
 
 -- some help in the beginning ;)
 data Knight = Knight
-    { knightHealth    :: Int
-    , knightAttack    :: Int
-    , knightEndurance :: Int
+    { knightHealth     :: Int
+    , knightAttack     :: Int
+    , knightEndurance  :: Int
+    , knightExperience :: Int
+    , knightGold       :: Int
+    , knightSword      :: Maybe Sword
+    , knightShield     :: Maybe Shield
+    , knightAlive      :: Bool
+    }
+    deriving (Show)
+
+data Color
+    = Red
+    | Black
+    | Green
+    deriving (Show, Eq, Ord)
+
+{-
+type RedDragonChest = Chest Shield
+type BlackDragonChest = Chest Sword
+type GreenDragonChest = Chest
+-}
+
+data Sword = Sword
```

Comment:
Since it's a data type with exactly one constructor and exactly one field, it can be a `newtype` instead 🙂 
```suggestion
newtype Sword = Sword
```

Reply:
Oh, I'm still feeling not comfortable with datatypes. Will try this.